src/app/armory/weapon-roll-export.test.ts

```diff
@@ -0,0 +1,121 @@
+import { DimItem } from 'app/inventory/item-types';
+import { DestinySocketCategoryStyle } from 'bungie-api-ts/destiny2';
+import { formatWeaponRollForExport } from './weapon-roll-export';
+
+// Mock socket utils
+jest.mock('app/utils/socket-utils', () => ({
+  getWeaponArchetypeSocket: jest.fn(),
+  getSocketsWithStyle: jest.fn(),
+  isWeaponMasterworkSocket: jest.fn(),
+}));
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+const { getWeaponArchetypeSocket, getSocketsWithStyle, isWeaponMasterworkSocket } =
+  jest.requireMock('app/utils/socket-utils');
+
+describe('formatWeaponRollForExport', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return empty string for non-weapon items', () => {
+    const item = {
+      bucket: { inWeapons: false },
+      sockets: {},
+    } as DimItem;
+
+    const result = formatWeaponRollForExport(item);
+    expect(result).toBe('');
+  });
+
+  it('should return empty string for items without sockets', () => {
+    const item = {
+      bucket: { inWeapons: true },
+      sockets: null,
+    } as DimItem;
+
+    const result = formatWeaponRollForExport(item);
+    expect(result).toBe('');
+  });
```

Comment:
We shouldn't test these cases - instead, we should make sure to only show the button when there is a reasonable roll to copy.